src/main/resources/assets/computercraft/lua/rom/programs/shell.lua

```diff
@@ -19,6 +19,7 @@ local function createShellEnv( sDir )
     local tEnv = {}
     tEnv[ "shell" ] = shell
     tEnv[ "multishell" ] = multishell
+    shellenv = {}
```

Comment:
This should be declared outside this function, and should be local. You should probably have it be inherited from the parent shell too - like the the path and aliases.